flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/DefaultSchemaResolver.java

```diff
@@ -248,11 +252,11 @@ private void validateTimeColumn(String columnName, List<Column> columns) {
     }
 
     private void validateWatermarkExpression(LogicalType watermarkType) {
-        if (!hasRoot(watermarkType, LogicalTypeRoot.TIMESTAMP_WITHOUT_TIME_ZONE)
+        if (!hasFamily(watermarkType, LogicalTypeFamily.TIMESTAMP)
```

Comment:
with this you also allow timestamp's with time zone, we should be more precise here